project_checker.py

```diff
@@ -149,13 +124,18 @@ def __init__(self, project: QgsProject) -> None:
                 "fn": self.check_layer_configured,
                 "scope": None,
             },
+            {
+                "type": Feedback.Level.WARNING,
+                "fn": self.check_layer_package_prevention,
+                "scope": None,
+            },
         ]
 
     def check(self, scope: ExportType = None) -> ProjectCheckerFeedback:
         checked_feedback = ProjectCheckerFeedback()
 
         for check in self.project_checks:
-            if check["scope"] != scope:
+            if scope and check["scope"] and check["scope"] != scope:
```

Comment:
Is this meant to be `'scope' in check` or really a truthness check for `check["scope"]`?

Reply:
`scope` was marked as NULL-able, but now I made it non-null. So the condition is simplified to:

`if check["scope"] and check["scope"] != scope`, since `check["scope"]` might be `None`.